### PR TITLE
Add policy claim support for JWT

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Add policy claim support for JWT

## Motivation and Context
This way temporary credentials can use canned
policies on the server without configuring OPA.

## Regression
No

## How Has This Been Tested?
- This is tested by configuring `policy` claims in WSO2 follow the documentation updated in this PR to add new `policy` claim in the WSO2 console.

- Once added for the client grants service provider this is available via JWT which can be used by the server to validate the policies against OPA.

To create canned policies https://docs.minio.io/docs/minio-admin-complete-guide.html#policies use `mc`. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
Can this be reviewed @krishnasrinivas @poornas ?

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
Can this be reviewed @krishnasrinivas @poornas

**Comment by krishnasrinivas** [comment]
@harshavardhana yes will do

**Review by poornas — APPROVED** [review]
LGTM

File: cmd/iam.go [hunk 1]
```diff
@@ -269,12 +269,46 @@ func (sys *IAMSys) DeleteUser(accessKey string) error {
...
+		p, ok := sys.iamCannedPolicyMap[policyName]
+		if !ok {
+			return errInvalidArgument
+		}
+		if p.IsEmpty() {
+			delete(sys.iamPolicyMap, accessKey)
+			return nil
+		}
+
+		data, err := json.Marshal(policyName)
```
Comment (krishnasrinivas) [inline cmd/iam.go:294]: data, err := json.Marshal(p)

Comment (harshavardhana) [reply cmd/iam.go:294]: No the policyName is correct, canned policies are already available on the etcd in a separate location. This is just to store the canned name. @krishnasrinivas

**Review by krishnasrinivas — APPROVED** [review]

**Pull request merged**